parser/lexer.go

```diff
@@ -527,6 +530,9 @@ func (s *Scanner) scanString() (tok int, pos Pos, lit string) {
 			ch0 = handleEscape(s)
 		}
 		mb.writeRune(ch0, s.r.w)
+		if s.r.eof() {
```

Comment:
panic is caused by `s.r.inc()` use a out of range offset, use `if !s.r.eof() ` to pretent this is more readable. maybe?